topics/error-handling.md

```diff
@@ -0,0 +1,132 @@
+# Error handling guidelines
+
+Almost all errors in Cartridge follow `return nil, err` style, where
+`err` is an error object produced by `errors`
+[module](https://github.com/tarantool/errors). Cartridge doesn't raise
+errors except for bugs and functions contracts mismatch. Developing new
+roles should follow these guidelines as well.
+
+## Error objects in Lua
+
+Error classes help to locate the problem's source. For this purpose, an
+error object contains it's class, stack traceback and a message.
+
+```lua
+local errors = require('errors')
+local DangerousError = errors.new_class("DangerousError")
+
+local function some_fancy_function()
+
+    local something_bad_happens = true
+
+    if something_bad_happens then
+        return nil, DangerousError:new("Oh boy")
+    end
+
+    return "success" -- no
+end
+
+print(some_fancy_function())
+```
+
+```
+nil	DangerousError: Oh boy
+stack traceback:
+	test.lua:9: in function 'some_fancy_function'
+	test.lua:15: in main chunk
+```
+
+For uniform error handling `errors` provides `pcall` API:
```

Comment:
Please state explicitly, that errors `:pcall` API differs from what we have [in vanilla lua](https://www.lua.org/pil/8.4.html) (this is important). It also be good to mention, that for errors `:pcall` there is no difference between `return nil, err` and `error(err)` approaches.